staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -569,9 +569,14 @@ func (p *patcher) patchResource(ctx context.Context, scope *RequestScope) (runti
 	default:
 		return nil, false, fmt.Errorf("%v: unimplemented patch type", p.patchType)
 	}
+	dedupOwnerReferencesTransformer := func(_ context.Context, obj, _ runtime.Object) (runtime.Object, error) {
+		// Dedup owner references after mutating admission happens
+		dedupOwnerReferencesAndAddWarning(obj, ctx, true)
+		return obj, nil
+	}
 
 	wasCreated := false
-	p.updatedObjectInfo = rest.DefaultUpdatedObjectInfo(nil, p.applyPatch, p.applyAdmission)
+	p.updatedObjectInfo = rest.DefaultUpdatedObjectInfo(nil, p.applyPatch, p.applyAdmission, dedupOwnerReferencesTransformer)
```

Comment:
managed fields get updated in `applyPatch`. Currently a bad request can still wipe out managed fields.

Reply:
interesting how you mention that at the exact same time that this happened: https://github.com/kubernetes/kubernetes/issues/96688

Reply:
cc @kwiesmueller

Reply:
Wouldn't it be enough to also dedup before apply?
Well... we'd have to dedup on the patchObj we decode into unstructured I think.

In general, I think Apply would still be broken of sorts until we make all the fields you equal on keys.
This could even solve both problems because at least for apply the request would be rejected on duplicate keys?
Correct me if I'm wrong @apelisse

Reply:
Apply does the schema validation so there's no way you're going to be able to apply duplicated keys. If you apply a key that is already in the object, it's going to be merged with the applied key. And, unfortunately, if the object has duplicated keys, I don't know if it can be applied.

Reply:
here I'm thinking of: 
- if a patch appends a duplicate owner ref, the managedFields update will fail

mutating admission is not involved. But I agree #96688 is also a problem

Reply:
Yeah what I meant is, that we should make all fields in the ownerRef keys in the schema so we replicate the same behavior?

Reply:
> Apply does the schema validation so there's no way you're going to be able to apply duplicated keys

That's good news. So the problem I was thinking of https://github.com/kubernetes/kubernetes/pull/96185#discussion_r526397606 only happens to the other patchers (json patch, json merge patch, SMP)